scripts/docker-cli-monitor.ts

```diff
@@ -0,0 +1,82 @@
+import path from 'path';
+
+import { readDependencyVersions, getOctokit, RancherDesktopRepository, IssueOrPullRequest } from 'scripts/lib/dependencies';
+import semver from 'semver';
+
+const GITHUB_OWNER = process.env.GITHUB_REPOSITORY?.split('/')[0] || 'rancher-sandbox';
+const GITHUB_REPO = process.env.GITHUB_REPOSITORY?.split('/')[1] || 'rancher-desktop';
+const DOCKER_CLI_OWNER = process.env.DOCKER_CLI_OWNER || 'docker';
+const DOCKER_CLI_REPO = process.env.DOCKER_CLI_REPO || 'cli';
+const TAG_REGEX = /^v[0-9]+\.[0-9]+\.[0-9]+$/;
+const SCRIPT_NAME = 'docker-cli-monitor';
+const mainRepo = new RancherDesktopRepository(GITHUB_OWNER, GITHUB_REPO);
+
+async function getLatestDockerCliVersion(): Promise<string> {
+  const result = await getOctokit().rest.repos.listTags({
+    owner: DOCKER_CLI_OWNER, repo: DOCKER_CLI_REPO, per_page: 100,
+  });
+  const tags = result.data;
+  const fullReleaseTags = tags.filter(tag => TAG_REGEX.test(tag.name));
+
+  if (fullReleaseTags.length === 0) {
+    throw new Error('Failed to find any valid tags');
+  }
+  fullReleaseTags.sort((previous, current) => {
+    const previousWithoutV = previous.name.replace('v', '');
+    const currentWithoutV = current.name.replace('v', '');
+
+    return semver.rcompare(previousWithoutV, currentWithoutV, { loose: true });
+  });
+  const latestTag = fullReleaseTags[0];
+
+  return latestTag.name;
```

Comment:
```suggestion
  return fullReleaseTags.shift().name;
```
If we use `semver.compare` instead of `semver.rcompare`, then it's `fullReleaseTags.pop().name` (which is slightly cheaper).

Thinking about it, though, we don't need to sort the whole thing, just the max, so…
```ts
const rawVersions = fullReleaseTags.map(r => r.name);
return rawVersions.reduce((p, v) => semver.gt(p.replace('v', ''), v.replace('v', ''), {loose: true}) ? p : v);
```

Reply:
I went to use `fullReleaseTags.pop().name`, but typescript is telling me that `fullReleaseTags.pop()` could be `undefined`. So we'd have to introduce a check for this. I think we'll just leave it, thanks for the feedback though.